src/main/java/ch/njol/skript/entity/BoatChestData.java

```diff
@@ -0,0 +1,145 @@
+import org.bukkit.entity.ChestBoat;
+import org.bukkit.inventory.ItemStack;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.Random;
+
+public class BoatChestData extends EntityData<ChestBoat> {
+
+	private static ItemType oakBoat = null;
+	private static ItemType spruceBoat = null;
+	private static ItemType birchBoat = null;
+	private static ItemType jungleBoat = null;
+	private static ItemType acaciaBoat = null;
+	private static ItemType darkOakBoat = null;
+
+	static {
+		if (Skript.classExists("org.bukkit.entity.ChestBoat")) {
+			oakBoat = Aliases.javaItemType("oak chest boat");
+			spruceBoat = Aliases.javaItemType("spruce chest boat");
+			birchBoat = Aliases.javaItemType("birch chest boat");
+			jungleBoat = Aliases.javaItemType("jungle chest boat");
+			acaciaBoat = Aliases.javaItemType("acacia chest boat");
+			darkOakBoat = Aliases.javaItemType("dark oak chest boat");
+			EntityData.register(BoatChestData.class, "chest boat", ChestBoat.class, 0,
+				"chest boat", "any chest boat", "oak chest boat", "spruce chest boat", "birch chest boat",
+				"jungle chest boat", "acacia chest boat", "dark oak chest boat");
+		}
+	}
+
+	public BoatChestData() {
+		this(0);
+	}
+
+	public BoatChestData(@Nullable TreeSpecies type) {
+		this(type != null ? type.ordinal() + 2 : 1);
+	}
+
+	private BoatChestData(int type) {
+		matchedPattern = type;
+	}
+
+	@Override
+	protected boolean init(Literal<?>[] exprs, int matchedPattern, SkriptParser.ParseResult parseResult) {
+
+		return true;
+	}
+
+	@Override
+	protected boolean init(@Nullable Class<? extends ChestBoat> c, @Nullable ChestBoat e) {
+		if (e != null)
+			matchedPattern = 2 + e.getWoodType().ordinal();
+		return true;
+	}
+
+	@Override
+	public void set(ChestBoat entity) {
+		if (matchedPattern == 1) // If the type is 'any boat'.
+			matchedPattern += new Random().nextInt(TreeSpecies.values().length); // It will spawn a random boat type in case is 'any boat'.
+		if (matchedPattern > 1) // 0 and 1 are excluded
+			entity.setWoodType(TreeSpecies.values()[matchedPattern - 2]); // Removes 2 to fix the index.
+	}
+
+	@Override
+	protected boolean match(ChestBoat entity) {
+		return matchedPattern <= 1 || entity.getWoodType().ordinal() == matchedPattern - 2;
+	}
+
+	@Override
+	public Class<? extends ChestBoat> getType() {
+		return ChestBoat.class;
+	}
+
+	@Override
+	public EntityData getSuperType() {
+		return new BoatChestData(matchedPattern);
+	}
+
+	@Override
+	protected int hashCode_i() {
+		return matchedPattern <= 1 ? 0 : matchedPattern;
+	}
+
+	@Override
+	protected boolean equals_i(EntityData<?> obj) {
+		if (obj instanceof BoatData)
+			return matchedPattern == ((BoatData) obj).matchedPattern;
+		return false;
+	}
+
+	@Override
+	public boolean isSupertypeOf(EntityData<?> e) {
+		if (e instanceof BoatData)
+			return matchedPattern <= 1 || matchedPattern == ((BoatData) e).matchedPattern;
+		return false;
+	}
+
+
+	public boolean isOfItemType(ItemType itemType) {
```

Comment:
Is this method overriden from somewhere?

Reply:
ah yes, in DefaultComparators
Thanks for noticing that. I'll fix that up later today

Reply:
ok this has been fixed :)